tee-worker/omni-executor/rpc-server/src/methods/submit_aes_request.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+	authentication::{
+		verify_auth_token_authentication, verify_email_authentication,
+		verify_oauth2_authentication, verify_web3_authentication, Authentication,
+	},
+	error_code::*,
+	request::{AesRequest, DecryptableRequest},
+	server::RpcContext,
+	utils::hex::{FromHexPrefixed, ToHexPrefixed},
+};
+use executor_core::native_call::NativeCall;
+use jsonrpsee::{
+	types::{ErrorCode, ErrorObject},
+	RpcModule,
+};
+use native_task_handler::NativeTask;
+use parentchain_primitives::{Nonce, OmniAccountAuthType};
+use parity_scale_codec::{Decode, Encode};
+use std::{fmt::Debug, sync::Arc};
+use tokio::sync::oneshot;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub struct AuthenticatedCall {
+	pub call: NativeCall,
+	pub nonce: Nonce,
+	pub authentication: Authentication,
+}
+
+pub fn register_submit_aes_request(module: &mut RpcModule<RpcContext>) {
+	module
+		.register_async_method("native_submitAesRequest", |params, ctx, _| async move {
```

Comment:
Would you add another `native_submitPlainRequest` which doesn't need to be encrypted with aes?

Also a question:

rpc methods start with `native_` stand for a nativeCall request, right? A `NativeCall` isn't equal to `AuthenticatedCall`?

Like we should differentiate between aes-encrypted `nativeCall`, and authenticated `nativeCall`, and even aes-encrypted authenticated `nativeCall`:
- aes means we want the payload to be encrypted/private
- authenticated means we want to verify the origin of the requester

I'm wondering if there're better structure/names to make these more readable

Reply:
The `AuthenticatedCall` is the equivalent of `TrustedCallSigned` or `TrustedCallAuthenticated` in the identity worker and `NativeCall` is the equivalent of `TrustedCall`. 

How about if we call them `NativeCallAuthenticated` and `NativeCall`? 

Now that you mentioned this about the encrypted requests, why do we need them for direct calls? Would `native_submitNativeCallAuthenticated` be enough for these calls? The client would only need to send the `AuthenticatedNativeCall` encoded.

Also, here we are skipping another wrapper (`TrustedOperation`). If it's OK to remove the need for encrypting the requests for direct calls it would simplify even more the client side. What do you think @Kailai-Wang ?